ouroboros-consensus/bench/PerasCertDB-bench/Main.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE LambdaCase #-}
+
+-- | This module contains benchmarks for Peras chain weight calculation as implemented by
+--   the by the 'Ouroboros.Consensus.Storage.PerasCertDB.API.boostedWeightForFragment'
+--   function.
+--
+--   We benchmark the calculation on a static sequence of chain fragments of increasing
+--   length, ranging from 0 to around 2160, with a sampling rate of 100. The chain fragments
+--   are instantiated with 'TestBlock', and every 300 blocks there is a booster block with
+--   weight 15. All parameters are set in 'benchmarkParams'.
+module Main (main) where
+
+import Data.List (iterate')
+import Data.Map.Strict qualified as Map
+import Numeric.Natural (Natural)
+import Ouroboros.Consensus.Block (PerasWeight (PerasWeight))
+import Ouroboros.Consensus.Storage.PerasCertDB.API
+  ( PerasWeightSnapshot (..)
+  , boostedWeightForFragment
+  )
+import Ouroboros.Network.AnchoredFragment qualified as AF
+import Test.Tasty.Bench
+import Test.Util.TestBlock (TestBlock)
```

Comment:
Hmm, for purely comparative measurements, `Test.Util.TestBlock` should be fine, but for absolute estimations (useful for deciding how much we need to optimize this stuff), it is a bit degenerate, eg because `HeaderHash TestBlock ~ NonEmpty Word64` which is pretty different from what Cardano blocks use as a header hash, namely (short) bytestrings.

`Test.Ouroboros.Storage.TestBlock` is a bit more realistic (it uses `Int`s as hashes); we could also define a super simple benchmark block here, but we might need to implement a bunch of classes.